pymeasure/instruments/agilent/agilentB1500.py

```diff
@@ -1567,39 +1550,29 @@ def __init__(self, smu_type):
 ######################################
 
 
-class SPGU:
+class SPGU(Channel):
     """Provide specific methods for the SPGU module of the Agilent B1500 mainframe
 
     :param AgilentB1500 parent: Instance of the B1500 mainframe class
     :param int channel: Channel number of the SPGU
     """
 
-    def __init__(self, parent, channel, name, **kwargs):
-        # to allow garbage collection for cyclic references
-        self._b1500 = weakref.proxy(parent)
-        channel = strict_discrete_set(channel, range(1, 5))
-        self.channel = channel
-        self.name = name
-        self.ch1 = SPGUChannel(self, int(str(self.channel) + "01"))
-        self.ch2 = SPGUChannel(self, int(str(self.channel) + "02"))
-
-    def write(self, string):
-        """Wrap :meth:`.Instrument.write` method of B1500."""
-        self._b1500.write(string)
-
-    def ask(self, string):
-        """Wrap :meth:`~.Instrument.ask` method of B1500."""
-        return self._b1500.ask(string)
-
-    def start_output(self):
-        """Start SPGU output (``SRP``)"""
-        self.write("SRP")
-
-    def stop_output(self):
-        """Stop SPGU output (``SPP``). The channel outputs the base voltage."""
-        self.write("SPP")
+    def __init__(self, parent, channel, **kwargs):
+        super().__init__(parent, channel, **kwargs)
+        self.ch1 = self.add_child(SPGUChannel, int(str(self.id) + "01"), prefix="ch")
+        self.ch2 = self.add_child(SPGUChannel, int(str(self.id) + "02"), prefix="ch")
```

Comment:
Just a note for string manipulation: `f"{self.id}01"` will work instead of `str(self.id) + "01"`.
Those f-strings are quite powerful.